include/pipe.h

```diff
@@ -0,0 +1,41 @@
+#include <mutex.h>
+#include <condvar.h>
+#include <thread.h>
+#include <file.h>
+#include <uio.h>
+#include <vnode.h>
+
+/* size of pipe buffer */
+#define PIPE_SIZE 4096
+
+#define PIPE_EOF 0x1
+
+typedef struct pipebuf {
```

Comment:
There's already an implementation of [ringbuf](https://github.com/cahirwpz/mimiker/blob/master/include/ringbuf.h) in our code base. Would it be possible to merge `pipebuf` into `ringbuf` and have single implementation?